src/formnetworks/abstractformnetwork.jl

```diff
@@ -69,11 +74,12 @@ end
 
 function environment(
   f::AbstractFormNetwork,
-  original_state_vertices::Vector;
+  original_vertices::Vector;
+  vertex_mapping_function=state_vertices,
   alg=default_environment_algorithm(),
   kwargs...,
 )
-  form_vertices = state_vertices(f, original_state_vertices)
+  form_vertices = vertex_mapping_function(f, original_vertices)
```

Comment:
I see that is almost how it is designed, so I think my primary concern is using an if-statement.

Is it harmful to pass a `partitioned_vertices` keyword argument to every backend? Other contraction backends can ignore it if they want to, then this function would be generic to any `alg`. So my proposal would be:
```julia
function environment(
  f::AbstractFormNetwork,
  subvertices::Vector;
  alg=default_environment_algorithm(),
  kwargs...,
)
  partitioned_vertices = group(v -> original_state_vertex(f, v), vertices(f))
  return environment(
      tensornetwork(f), subvertices; alg, partitioned_vertices, kwargs...
    )
else
```
and it is up to the different backends to process `partitioned_vertices` as they see fit (and maybe ignore them if they want to).

Reply:
We don't necessarily have to address the redesign of the form types in this PR. We took a best guess at the design of the form networks a bit in a vacuum and now have to see how those design choices play out as we try to write more sophisticated code with it. I think this PR is already exposing some warts of the current design.

Reply:
Yeah we can remove the `if` statement. I was probably trying to avoid sending a "dud" `kwarg` down to `environment` when `alg = "exact"` but you're right that it looks a bit messy and isn't really necessary and there's nothing so bad about `partitioned_vertices` being passed down if its not needed.

We could always define `default_partitioned_vertices(alg, tn)` or something and have it return `nothing` if the `alg` is not a partitioning-based `alg`.